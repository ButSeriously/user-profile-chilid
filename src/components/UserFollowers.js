import PropTypes from 'prop-types';
import React from 'react';
import UserFollowersCounter from './UserFollowersCounter';

const UserFollowers = props => (
  <div className="user_followers">
    <div className="user_followers__stats">
      <UserFollowersCounter user_stats={props.likes} />
      <UserFollowersCounter user_stats={props.following} />
      <UserFollowersCounter user_stats={props.followers} />
    </div>
    <button className="user_followers__button" onClick={props.toggleFollowers}>Follow</button>
  </div>
);

UserFollowers.propTypes = {
  likes: PropTypes.array.isRequired,
  following: PropTypes.array.isRequired,
  followers: PropTypes.array.isRequired,
  toggleFollowers: PropTypes.func.isRequired,
};

export default UserFollowers;
