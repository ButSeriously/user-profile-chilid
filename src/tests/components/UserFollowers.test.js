import React from 'react';
import { shallow } from 'enzyme';
import UserFollowers from '../../components/UserFollowers';

test('should render UserFollowers', () => {
  const userStats = {
    likes: ['Likes', 300],
    following: ['Following', 70],
    followers: ['Followers', 1100],
  };
  const toggleFollowers = () => {};
  const wrapper = shallow(<UserFollowers {...userStats} toggleFollowers={toggleFollowers} />);
  expect(wrapper).toMatchSnapshot();
});
