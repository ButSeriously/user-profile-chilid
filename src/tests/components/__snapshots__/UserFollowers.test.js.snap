// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserFollowers 1`] = `
<div
  className="user_followers"
>
  <div
    className="user_followers__stats"
  >
    <UserFollowersCounter
      userStats={
        Array [
          "Likes",
          300,
        ]
      }
    />
    <UserFollowersCounter
      userStats={
        Array [
          "Following",
          70,
        ]
      }
    />
    <UserFollowersCounter
      userStats={
        Array [
          "Followers",
          1100,
        ]
      }
    />
  </div>
  <button
    className="user_followers__button"
    onClick={[Function]}
  >
    Follow
  </button>
</div>
`;
